data-science-onramp/ai-platform/sklearn_test.py

```diff
@@ -0,0 +1,92 @@
+import os
+import re
+import tarfile
+import time
+import uuid
+
+from google.cloud import storage
+from google.cloud.aiplatform import gapic as aip
+from googleapiclient import discovery
+
+from typing import Callable
+import pytest
+
+STAGING_BUCKET = f"sklearn-job-dir-{uuid.uuid4()}"
+INPUT_DIR = "inputs"
+TRAINER_DIR = "modules"
+MODEL_DIR = "model"
+TRAIN_DATA = "train.csv"
+TRAINER_TAR = "trainer.tar.gz"
+TRAIN_DATA_PATH = f"gs://{STAGING_BUCKET}/{INPUT_DIR}/{TRAIN_DATA}"
+
+PROJECT_ID = os.environ["GOOGLE_CLOUD_PROJECT"]
+REGION = "us-central1"
+MODEL_NAME = f"sklearn-test-{uuid.uuid4()}"
+JOB_ID = f"sklearn_{str(uuid.uuid4())[:7]}"
+DEPLOY_IMAGE = "gcr.io/cloud-aiplatform/training/scikit-learn-cpu.0-23:latest"
+
+TERMINAL_STATES = [
+    aip.JobState.JOB_STATE_SUCCEEDED,
+    aip.JobState.JOB_STATE_FAILED,
+    aip.JobState.JOB_STATE_CANCELLING,
+    aip.JobState.JOB_STATE_CANCELLED
+]
+
+@pytest.fixture(autouse=True)
+def setup_teardown():
+    storage_client = storage.Client()
+    bucket = storage_client.create_bucket(STAGING_BUCKET, location=REGION)
+    bucket.blob(f"{INPUT_DIR}/{TRAIN_DATA}").upload_from_filename(TRAIN_DATA, timeout=600)
+
+    with tarfile.open(TRAINER_TAR, mode="x:gz") as tar:
+        tar.add(f"{TRAINER_DIR}/")
+
+    bucket.blob(TRAINER_TAR).upload_from_filename(TRAINER_TAR)
+
+    yield bucket
+
+    # [blob.delete() for blob in bucket.list_blobs()]
```

Comment:
@bradmiro I think this might still be needed to delete the output model right?

Also, please delete the custom training pipeline here or the test will keep making more of them.

Reply:
Fixed!

Reply:
Looks good!